R/is_numeric_linter.R

```diff
@@ -97,7 +97,7 @@ is_numeric_linter <- function() {
       class_expr,
       source_expression = source_expression,
       lint_message = paste(
-        'is.numeric(x) is the same as class(x) %in% c("integer", "numeric").',
+        'Use is.numeric(x) instead of the equivalent class(x) %in% c("integer", "numeric").',
```

Comment:
Actually, this is factually incorrect.
``` r
tmp <- structure(1.0, class = "foo")
is.numeric(tmp)
#> [1] TRUE
class(tmp) %in% c("integer", "numeric")
#> [1] FALSE
```

<sup>Created on 2023-12-06 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
Good point. Just drop "the equivalent" then? "Use is.numeric() instead of class()..."?

Reply:
```suggestion
        'Use is.numeric(x) instead of class(x) %in% c("integer", "numeric"), or `inherits(x, "integer")` if you need to check classes.',
```

Reply:
I worry that's a distraction... needing `is.numeric(x) && inherits(x, c("integer", "numeric"))` should be pretty rare + advanced usage.

Reply:
That's true, and it's at the end of the lint message for this. I feel it adds value because it also teaches that you shouldn't use `class(x) %in% ...` in general.

Reply:
we've already got `class_equals_linter()` for that 😉

```r
lint("class(x) %in% c('integer', 'numeric')\n", class_equals_linter())
# <text>:1:1: warning: [class_equals_linter] Instead of comparing class(x) with %in%, use inherits(x, 'class-name') or is.<class> or is(x, 'class')
# class(x) %in% c('integer', 'numeric')
# ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Ah, thanks for pointing that out. Then LGTM 😁